pkg/graph/resolvers/system_intake_actions.go

```diff
@@ -236,3 +236,222 @@ func CreateSystemIntakeActionRequestEdits(
+	}
+
+	// No validity check needed:
+	// * Issuing this decision is valid in all steps
+	// * Issuing this decision is valid both when an intake is open and when it's closed (in the latter case, it's changing the decision)
+	// * Even if a rejection decision has already been issued, an admin can confirm that decision on a reopened intake through this action
+
+	// update workflow state
+	intake.Step = models.SystemIntakeStepDECISION
+	intake.State = models.SystemIntakeStateCLOSED
+	intake.DecisionState = models.SIDSNotApproved
+
+	// update other fields
+	intake.RejectionReason = null.StringFrom(input.Reason)
+	intake.DecisionNextSteps = null.StringFrom(input.NextSteps)
+	intake.TRBFollowUpRecommendation = &input.TrbFollowUp
+
+	updatedTime := time.Now()
+	intake.UpdatedAt = &updatedTime
+
+	// All the different database calls aren't in a single atomic transaction;
+	// in the case of a system failure, some data from the action might be saved, but not all.
+	// As of this function's initial implementation, we're accepting that risk.
+	// If we create a general way to wrap several store methods calls in a transaction later, we can use that.
+
+	errGroup := new(errgroup.Group)
+	var updatedIntake *models.SystemIntake // declare this outside the function we pass to errGroup.Go() so we can return it
+
+	// save intake
+	errGroup.Go(func() error {
+		var errUpdateIntake error // declare this separately because if we use := on next line, compiler thinks we're declaring a new updatedIntake variable as well
+		updatedIntake, errUpdateIntake = store.UpdateSystemIntake(ctx, intake)
+		if errUpdateIntake != nil {
+			return errUpdateIntake
+		}
+
+		return nil
+	})
+
+	// save action (including additional info for email, if any)
+	errGroup.Go(func() error {
+		action := models.Action{
+			IntakeID:       &input.SystemIntakeID,
+			ActionType:     models.ActionTypeREJECT,
+			ActorName:      adminUserInfo.CommonName,
+			ActorEmail:     adminUserInfo.Email,
+			ActorEUAUserID: adminEUAID,
+			Step:           &intake.Step,
+		}
+		if input.AdditionalInfo != nil {
+			action.Feedback = null.StringFromPtr(input.AdditionalInfo)
+		}
+
+		_, errCreatingAction := store.CreateAction(ctx, &action)
+		if errCreatingAction != nil {
+			return errCreatingAction
+		}
+
+		return nil
+	})
+
+	// save admin note
+	if input.AdminNote != nil {
+		errGroup.Go(func() error {
+			adminNote := &models.SystemIntakeNote{
+				SystemIntakeID: input.SystemIntakeID,
+				AuthorEUAID:    adminEUAID,
+				AuthorName:     null.StringFrom(adminUserInfo.CommonName),
+				Content:        null.StringFromPtr(input.AdminNote),
+			}
+
+			_, errCreateNote := store.CreateSystemIntakeNote(ctx, adminNote)
+			if errCreateNote != nil {
+				return errCreateNote
+			}
+
+			return nil
+		})
+	}
+
+	if err := errGroup.Wait(); err != nil {
+		return nil, err
+	}
+
+	return updatedIntake, nil
+}
+
+// IssueLCID handles a Not Approved by GRB action on an intake as part of Admin Actions v2
```

Comment:
I think this comment should concern handling the issuance of an LCID, right?

Reply:
Fixed in https://github.com/CMSgov/easi-app/pull/2155/commits/8621a85ba048954dfbfedf99d072d8ae8e1bb2e8.